github/repos_rules.go

```diff
@@ -85,6 +85,21 @@ type RuleFileParameters struct {
 	RestrictedFilePaths *[]string `json:"restricted_file_paths"`
 }
 
+// RuleMaxFilePathLengthParameters represents the max_file_path_length rule parameters.
+type RuleMaxFilePathLengthParameters struct {
+	MaxFilePathLength int `json:"max_file_path_length"`
+}
+
+// RuleFileExtensionRestrictionParameters represents the file_extension_restriction rule parameters.
+type RuleFileExtensionRestrictionParameters struct {
+	RestrictedFileExtensions *[]string `json:"restricted_file_extensions"`
```

Comment:
Unless there is a strong reason to use a pointer to a slice, we try very hard in this repo to avoid it.
I'm not aware of a strong need in this circumstance, but please correct me if I'm wrong.

```suggestion
	RestrictedFileExtensions []string `json:"restricted_file_extensions"`
```

Reply:
I did wonder, took a look at the prior push rule addition which was a pointer. Wasn't sure if that was necessary either! will modify